src/plugins/data_source_management/public/components/create_data_source_wizard/components/create_form/create_data_source_form.tsx

```diff
@@ -365,7 +365,7 @@ export class CreateDataSourceForm extends React.Component<
                   }
                 )}
                 isInvalid={!!this.state.formErrorsByField.createCredential.username.length}
-                value={this.state.auth.credentials.username || ''}
+                value={this.state.auth.credentials.username?.toString() || ''}
```

Comment:
If we expect the text field input to be a string, maybe better to do this?
```typescript
value={typeof this.state.auth.credentials.username === 'string' ? this.state.auth.credentials.username : ''}
```
The same for `password` `region` `accessKey` and `secretKey`